### PR TITLE
Add rft to files

### PR DESCRIPTION
**Pull request by jonnyh404**
Reuses code from the add_operators to be able to add any request for data outputs. To make a fully functioning "add_rft" method we would need to be able to add to "WELLLIST" in the network file. #282

### CODE REVIEW

File: ResSimpy/Nexus/NexusReporting.py [hunk 1]
```diff
@@ -4,19 +4,39 @@
...
     """Class for handling output requests in Nexus."""
     date: str
     output: str
     output_type: OutputType
     output_frequency: FrequencyEnum
     output_frequency_number: None | float
 
+    def to_table_line(self, headers: list[str]) -> str:
+        """String representation of the single line within an Output request table."""
+        _ = headers
```
Comment (kwr3) [inline ResSimpy/Nexus/NexusReporting.py:25]: slightly confused on this statement.

File: ResSimpy/Nexus/NexusReporting.py [hunk 2]
```diff
@@ -4,19 +4,39 @@
...
+    def units(self) -> BaseUnitMapping:
+        return BaseUnitMapping(unit_system=None)
+
+    @staticmethod
+    def get_keyword_mapping() -> dict[str, tuple[str, type]]:
+        return {}
+
 
 @dataclass(kw_only=True)
 class NexusOutputContents:
```
Comment (kwr3) [inline ResSimpy/Nexus/NexusReporting.py:42]: doc string needed here?

File: ResSimpy/Nexus/NexusReporting.py [hunk 3]
```diff
@@ -223,3 +248,14 @@ def _get_output_contents(table_file_as_list: list[str], date: str, output_type:
             resulting_output_contents.append(output_contents_obj)
 
         return resulting_output_contents
+
+    def _add_array_output_request(self, output_request: NexusOutputRequest) -> None:
```
Comment (kwr3) [inline ResSimpy/Nexus/NexusReporting.py:252]: consider doc string

File: ResSimpy/Nexus/nexus_add_new_object_to_file.py [hunk 4]
```diff
@@ -169,9 +169,10 @@ def write_out_new_table_containing_object(self, obj_date: str,
...
         headers = [k for k, v in nexus_mapping.items() if v[0] in object_properties]
-        write_out_headers = ' '.join(headers)
-        new_table_as_list.append(write_out_headers)
-        new_table_as_list.append(new_obj.to_table_line(headers))
+        if headers:
+            write_out_headers = ' '.join(headers)
+            new_table_as_list.append(write_out_headers)
+        new_table_as_list.append(new_obj.to_table_line(headers=headers))
         new_table_as_list.append(self.table_footer)
         new_table_as_list = [x + '\n' if not x.endswith('\n') else x for x in new_table_as_list]
```
Comment (kwr3) [inline ResSimpy/Nexus/nexus_add_new_object_to_file.py:177]: nice list comprehension

File: ResSimpy/Nexus/NexusReporting.py (hunk 1 above)
Comment (jonnyh404) [reply ResSimpy/Nexus/NexusReporting.py:25]: the `_ = headers` is to dispose of the headers variable which is required to match the variables from the parent class but not required for this object.

**Review by cflynn3 — APPROVED** [review]
Looks good to me, please have a look at Keith's comments though.

**Pull request merged**